src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -1287,23 +1284,18 @@ bool TryFastTrackPrimitiveTypes(object obj, out string str)
                     case TypeCode.UInt32:
                     case TypeCode.Int64:
                     case TypeCode.UInt64:
-                        str = obj.ToString();
-                        break;
                     case TypeCode.DateTime:
-                        DateTime dt = (DateTime)obj;
-                        str = dt.ToString(formatProvider);
-                        break;
                     case TypeCode.Decimal:
-                        Decimal dec = (Decimal)obj;
-                        str = dec.ToString(formatProvider);
+                        var formattable = (IFormattable)value;
```

Comment:
Interesting! Can you clarify why we use the interface?
I see there is many types supported the interface
https://source.dot.net/#System.Private.CoreLib/shared/System/IFormattable.cs,09da57c2c21a3a44,references - Single, Double, Datetime, Int16/32/64 and so on. Make sense use the interface for them?

Reply:
> Can you clarify why we use the interface?

Because ... all those primitive types support `ToString(string format, IFormatProvider)` (since they implement `IFormattable`), and we want to honor the passed in `format` and `formatProvider` when they are not null.

> Single, Double, Datetime, Int16/32/64 and so on. Make sense use the interface for them?

I already do that in the code. Now they honor the `format` and `formatProvider` passed in.

Reply:
Ah, sorry - float is below.